recipes/ffmpeg/all/conanfile.py

```diff
@@ -313,23 +311,23 @@ def validate(self):
             for feature in features:
                 used = used or self.options.get_safe(feature)
             if not used:
-                raise ConanInvalidConfiguration("FFmpeg '{}' option requires '{}' option to be enabled".format(
-                    dependency, "' or '".join(features)))
+                dependency_string = "' or '".join(features)
+                raise ConanInvalidConfiguration(f"FFmpeg '{dependency_string}' option requires '{dependency_string}' option to be enabled")
 
     def build_requirements(self):
         if self.settings.arch in ("x86", "x86_64"):
-            self.build_requires("yasm/1.3.0")
-        self.build_requires("pkgconf/1.9.3")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+            self.tool_requires("yasm/1.3.0")
+        self.tool_requires("pkgconf/1.9.3")
+        if self._settings_build.os == "Windows" and not self.win_bash:
```

Comment:
```suggestion
        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
```

recipe is not migrated to conan v2 build helpers, so you have to keep this logic. Moreover testing self.win_bash in conan v2 is not correct, it's not set externally .